test/test_jit.py

```diff
@@ -6075,6 +6075,33 @@ def test_script_for_in_range_if_ast(x):
 
         self.assertEqual(test_script_for_in_range_if_ast(*inputs).shape[0], 20)
 
+    def test_script_for_in_range_start_end(self):
+        @torch.jit.script
```

Comment:
Don't annotate `fn`s you pass to checkScript with `@script`. checkScript does that internally and compares the output of the Python function to the script function